### PR TITLE
create reusable Button component and use it everywhere it's possible

### PR DESCRIPTION
**Pull request by PupS1k**

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+const Button = ({className, onClick, text, img, disabled}) => (
+  <button
+    type="button"
+    className={className}
+    onClick={onClick}
+    disabled={disabled}
+  >
+    {img && <img className={img.className} src={img.src} alt={img.alt}/>}
```
Comment (azhalkouski) [inline src/components/Button.js]: Look how Buttons are implemented and used in React Toolbox and Material UI libraries.

http://react-toolbox.io/#/components/button
https://github.com/react-toolbox/react-toolbox/blob/dev/components/button/Button.js

https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/Button/Button.js
https://material-ui.com/components/buttons/

File: src/components/Header/index.js [hunk 2]
```diff
@@ -20,19 +20,24 @@ const Header = () => (
...
-        <img className="img--plus" src={plus} alt="Plus" />
-        <p className="btn--add-ad__text btn--add-product__text-full">Подать объявление</p>
-        <p className="btn--add-ad__text btn--add-product__text-reduction">Объявление</p>
-      </button>
-      <Button />
-
+      <Button
+        className="btn--add-product btn-type-primary"
+        img={{src: plus, className: 'img--plus', alt: 'Plus'}}
+        text={{className: 'btn--add-ad__text btn--add-product__text-full', txt: 'Подать объявление'}}
```
Comment (azhalkouski) [inline src/components/Header/index.js]: I thought we agreed to move these hard coded classnames into Button component, didn't we?

File: src/components/Header/style.css [hunk 3]
```diff
@@ -120,7 +120,7 @@ header{
     background: #29ad5b;
     border: none;
 }
-.btn--add-product__text-reduction{
+.btn-type-secondary{
```
Comment (azhalkouski) [inline src/components/Header/style.css]: According to the BEM official docs `block-name__element-name_modifier-name`.

File: src/components/Button.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    {text && <p className={text.className}>{text.txt}</p>}
+  </button>
+);
+
```
Comment (azhalkouski) [inline src/components/Button.js]: Button.propTypes = {}

Button.defaultProps? = {}

File: src/components/Navigation/index.js [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
+import Button from '../Button';
 import lines from "../../images/lines.png";
 import shirt from "../../images/shirt.png";
 import './style.css';
 
 
 const Navigation = () =>(
     <nav className="navigation">
-      <button className="btn--all-categories">
+      <Button className="btn--all-categories">
```
Comment (azhalkouski) [inline src/components/Navigation/index.js]: Pass img and text via props

File: src/components/Header/index.js [hunk 6]
```diff
@@ -20,19 +20,20 @@ const Header = () => (
...
+      <Button className="btn--add-product">
         <img className="img--plus" src={plus} alt="Plus" />
         <p className="btn--add-ad__text btn--add-product__text-full">Подать объявление</p>
         <p className="btn--add-ad__text btn--add-product__text-reduction">Объявление</p>
-      </button>
-      <Button />
-
+      </Button>
+      <Button className="btn--log-In">
+        <p>Вход</p>
```
[the remark this answers is not in the document]
Comment (PupS1k) [reply src/components/Header/index.js]: fixed

File: src/components/MainContent/FilterPanel/index.js [hunk 7]
```diff
@@ -88,13 +89,15 @@ class FilterPanel extends PureComponent {
...
-          <button type="button" className="btn--show-result" onClick={this.handleFilters}>
+          <Button className="btn--show-result" onClick={this.handleFilters}>
+            <p>
               Показать результаты(
-            {correctProducts.length}
-            )
-          </button>
-          <button type="button" className="btn--reset-filters">Сбросить фильтры</button>
-          <button type="button" className="btn--save-search">Сохранить поиск</button>
+              {correctProducts.length}
```
[the remark this answers is not in the document]
Comment (PupS1k) [reply src/components/MainContent/FilterPanel/index.js]: fixed

File: src/components/Button.js (hunk 1 above)
Comment (PupS1k) [reply src/components/Button.js]: mission complete!

File: src/components/Header/index.js (hunk 2 above)
Comment (PupS1k) [reply src/components/Header/index.js]: fixed

File: src/components/Navigation/index.js (hunk 5 above)
Comment (PupS1k) [reply src/components/Navigation/index.js]: fixed

File: src/components/Button.js (hunk 4 above)
Comment (PupS1k) [reply src/components/Button.js]: done

File: src/components/Header/style.css (hunk 3 above)
Comment (PupS1k) [reply src/components/Header/style.css]: rename className

File: src/components/Navigation/index.js [hunk 8]
```diff
@@ -1,37 +1,60 @@
...
+        <Button
+          className="nav-category "
+          icon={shirt}
+          alt="Shirt"
+          label="Модное лето"
+          iconSize="primary"
+          labelSize="large"
+          status="default"
+        />
+        {navBtn.map(text => (
```
Comment (azhalkouski) [inline src/components/Navigation/index.js]: Not as a change request but as a thing to thinks about...

ButtonGroup component :)

File: src/components/Footer/index.js [hunk 9]
```diff
@@ -10,25 +10,43 @@ import instagram from '../../images/instagram.png';
...
-          <img className="img--gistogram" src={gistogram} alt="Gistogram" />
-          <p>Реклама</p>
-        </a>
-        <a className="additional-link">Рассрочка</a>
-        <a className="additional-link">Доставка</a>
-        <a className="additional-link">Правила</a>
-        <a className="additional-link">Помощь</a>
-        <a className="additional-link">Выбрать регион</a>
+        <Button
+          className="additional-link "
```
[the remark this answers is not in the document]
Comment (PupS1k) [reply src/components/Footer/index.js]: fixed

File: src/components/Button.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    </button>
+  );
+};
+
+Button.propTypes = {
+  className: PropTypes.string,
+  label: PropTypes.string.isRequired,
+  labelSize: PropTypes.string.isRequired,
+  iconSize: PropTypes.string,
+  status: PropTypes.string.isRequired,
```
[the remark this answers is not in the document]
Comment (PupS1k) [reply src/components/Button.js]: fixed

File: src/components/Button.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+const Button = ({
+    className, onClick, label, icon, status, labelSize, iconSize, disabled, alt, bold
+  }) => {
+  const classNameBtn = 'btn btn_' + status;
+  const classNameIcon = 'btn__img btn__img_' + iconSize;
+  const classNameText = 'btn__text_' + labelSize +(bold ? ' btn__text_bold' : '');
+  return (
+    <button
+      type="button"
+      className={className ? className + classNameBtn : classNameBtn}
```
[the remark this answers is not in the document]
Comment (PupS1k) [reply src/components/Button.js]: install classNames and used

File: src/components/Button.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Button = ({
+    className, onClick, label, icon, status, labelSize, iconSize, disabled, alt, bold
+  }) => {
+  const classNameBtn = 'btn btn_' + status;
+  const classNameIcon = 'btn__img btn__img_' + iconSize;
+  const classNameText = 'btn__text_' + labelSize +(bold ? ' btn__text_bold' : '');
```
[the remark this answers is not in the document]
Comment (PupS1k) [reply src/components/Button.js]: fixed

**Pull request merged**